applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -583,13 +583,16 @@ void DistanceModificationProcess::DeactivateFullNegativeElements()
 
 void DistanceModificationProcess::SetContinuousDistanceToSplitFlag()
 {
-    #pragma omp parallel for
+
+    std::size_t nodes_per_element = mrModelPart.ElementsBegin()->GetGeometry().PointsNumber();
+    std::vector<double> elem_dist(nodes_per_element, 0.0);
+    #pragma omp parallel for firstprivate(elem_dist)
     for (int i_elem = 0; i_elem < static_cast<int>(mrModelPart.NumberOfElements()); ++i_elem) {
         auto it_elem = mrModelPart.ElementsBegin() + i_elem;
         auto &r_geom = it_elem->GetGeometry();
-        std::vector<double> elem_dist;
+        if(elem_dist.size() != r_geom.PointsNumber()) elem_dist.resize(r_geom.PointsNumber());
```

Comment:
I think this can never happen as the distance calculation only supports simplex elements.
```suggestion
```

Reply:
It was for keeping similar behavior as it was previously as a sefety check, but could be removed probably.

Reply:
I think that in this case we can assume that there is no "mixture" of elements.

Reply:
if we are sure we only want to use this with tets maybe we can directly use an array_1d<double,4> for the elem_dist and remove this first if (if NumberOfElements > 0 ...)

Reply:
Tets and triangles... Though we could use always the 4-component array I think it's a bit hacky.... I'd leave it as it is but without the check as suggested.